test/gtest/lstm_extra.cpp

```diff
@@ -0,0 +1,204 @@
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-hy --no-cy"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-dcx"},
+        std::pair{env, flags + commonFlags + " -dir-mode 1 --no-cy --no-dcx"},
+        std::pair{
+            env,
+            flags + commonFlags +
+                " -dir-mode 0 --no-hx --no-dhy --no-cx --no-dcy --no-hy --no-dhx --no-cy --no-dcx"},
+        std::pair{env,
+                  flags + commonFlags +
+                      " -dir-mode 1 --no-hx --no-dhy --no-cx --no-dcy --no-hy --no-dhx --no-cy "
+                      "--no-dcx"}};
+}
+
+using TestCase = decltype(GetTestCases({}))::value_type;
+
+class ConfigWithFloat : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+class ConfigWithHalf : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+bool IsTestSupportedForDevice()
+{
+    using namespace miopen::debug;
+    using e_mask = enabled<Gpu::gfx94X, Gpu::gfx103X, Gpu::gfx110X>;
+    using d_mask = disabled<Gpu::gfx900, Gpu::gfx906, Gpu::gfx908, Gpu::gfx90A>;
+    return miopen::debug::IsTestSupportedForDevice<d_mask, e_mask>();
+}
+
+void Run2dDriver(miopenDataType_t prec)
+{
+    if(!IsTestSupportedForDevice())
+    {
+        GTEST_SKIP();
+    }
+    std::vector<std::pair<EnvType, std::string>> params;
+    switch(prec)
+    {
+    case miopenFloat: params = ConfigWithFloat::GetParam(); break;
+    case miopenHalf: params = ConfigWithHalf::GetParam(); break;
+    case miopenFloat8:
+    case miopenBFloat8:
+    case miopenInt8:
+    case miopenBFloat16:
+    case miopenInt32:
+    case miopenDouble:
+        FAIL() << "miopenInt8, miopenBFloat16, miopenInt32, "
+                  "miopenDouble, miopenFloat8, miopenBFloat8 "
+                  "data types not supported by "
+                  "lstm_extra test";
+
+    default: params = ConfigWithFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value.second, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(), tokens.end(), std::back_inserter(ptrs), [](const auto& str) {
+            return str.data();
+        });
+        SetEnv(test_value.first);
+        if(Skip(prec))
+        {
+            GTEST_SKIP();
+        }
+        testing::internal::CaptureStderr();
+        test_drive<lstm_driver>(ptrs.size(), ptrs.data());
+        auto capture = testing::internal::GetCapturedStderr();
+        std::cout << capture;
+    }
+};
+
+} // namespace lstm_extra
+using namespace lstm_extra;
+
+TEST_P(ConfigWithFloat, FloatTest_lstm_extra) { Run2dDriver(miopenFloat); };
+
+TEST_P(ConfigWithHalf, HalfTest_lstm_extra) { Run2dDriver(miopenHalf); };
+
+INSTANTIATE_TEST_SUITE_P(LstmExtra, ConfigWithFloat, testing::Values(GetTestCases("--float")));
+
+INSTANTIATE_TEST_SUITE_P(LstmExtra, ConfigWithHalf, testing::Values(GetTestCases("--half")));
```

Comment:
Let's concentrate on functionally correct reworking and get rid of HALF testing in this PR.

Reply:
[resolved]